agent_status/scripts/controllers/AgentSummaryController.js

```diff
@@ -173,4 +177,142 @@ mainApp.controller('AgentSummaryController', function ($scope, $state, $timeout,
         });
     };
     $scope.getProfileDetails();
+
+    var setResourceData = function (profile) {
+        if (profile.slotState == 'Reserved') {
+            $scope.ReservedProfile.push(profile);
+        } else if (profile.other == 'Break') {
+            $scope.BreakProfile.push(profile);
+        }
+        else if (profile.slotState == 'Connected') {
+            $scope.ConnectedProfile.push(profile);
+        } else if (profile.slotState == 'AfterWork') {
+            $scope.AfterWorkProfile.push(profile);
+        } else if (profile.slotMode == 'Outbound' && profile.other == null) {
+            $scope.OutboundProfile.push(profile);
+        } else if (profile.slotState == 'Suspended') {
+            $scope.SuspendedProfile.push(profile);
+        } else if (profile.slotState == 'Available') {
+            $scope.AvailableProfile.push(profile);
+        } else {
+            $scope.profile.push(profile);
+        }
+    };
+
+    var removeExistingResourceData = function (profile) {
+        var stopSearch = false;
+        $scope.ReservedProfile.forEach(function (data, i) {
+            if(data.resourceName === profile.resourceName){
+                $scope.ReservedProfile.splice(i, 1);
+                stopSearch = true;
```

Comment:
Search and remove items form all arrays because there might be multiple item can be added due to concurrency, prevent that been happen